app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt

```diff
@@ -39,35 +39,38 @@ class MeetingRequestService : FirebaseMessagingService() {
     when (title) {
       "MEETING REQUEST" -> {
         val name = remoteMessage.data["senderName"] ?: "null"
-        MeetingRequestManager.meetingRequestViewModel?.addToMeetingRequestInbox(senderUid, message)
-        MeetingRequestManager.meetingRequestViewModel?.updateInboxOfMessagesAndThen() {}
+        MeetingRequestManager.meetingRequestViewModel?.addToMeetingRequestInbox(
+            senderUid, message) {
+              MeetingRequestManager.meetingRequestViewModel?.updateInboxOfMessages() {}
+            }
         showNotification(MSG_REQUEST, "from : $name")
       }
       "MEETING RESPONSE" -> {
         val name = remoteMessage.data["senderName"] ?: "null"
         val accepted = remoteMessage.data["accepted"]?.toBoolean() ?: false
         val senderToken = remoteMessage.data["senderToken"] ?: "null"
 
-        MeetingRequestManager.meetingRequestViewModel?.removeFromMeetingRequestSent(senderUid)
+        MeetingRequestManager.meetingRequestViewModel?.removeFromMeetingRequestSent(senderUid) {
+          MeetingRequestManager.meetingRequestViewModel?.updateInboxOfMessages() {}
+        }
         if (accepted) {
           showNotification(name + MSG_RESPONSE_ACCEPTED, "")
-          MeetingRequestManager.meetingRequestViewModel?.setMeetingConfirmation(
-              targetToken = senderToken,
-              newMessage = "The meeting with ${MeetingRequestManager.ourName} is confirmed !")
-          MeetingRequestManager.meetingRequestViewModel?.sendMeetingConfirmation()
+          MeetingRequestManager.meetingRequestViewModel?.addPendingLocation(senderUid) {}
         } else {
           showNotification(name + MSG_RESPONSE_REJECTED, "")
-          MeetingRequestManager.meetingRequestViewModel?.setMeetingConfirmation(
-              targetToken = senderToken,
-              newMessage = "The meeting with ${MeetingRequestManager.ourName} is cancelled !")
         }
       }
       "MEETING CONFIRMATION" -> {
         val name = remoteMessage.data["senderName"] ?: "null"
```

Comment:
You get the smalls values from remoteMessage.data several times, you can extract them once into variables : val senderUid = remoteMessage.data["senderUID"] ?: "null"
val senderName = remoteMessage.data["senderName"] ?: "null"
val message = remoteMessage.data["message"] ?: "null"